### DISCUSSION TITLE
Nature Instance requires prefab from project not itself if it's a prefab?

### DISCUSSION THREAD

**Discussion by AlexandreHurbe** [visualdesigncafe/nature-renderer#35] [Help]
Hi, I'm trying to use Nature Instances for some hand placed grass. I created the prefab with the Nature Instance component and then put it in my project folder. For the Nature Instance prefab I used itself (gameobject) as the prefab. However, when I place it in the scene, it give's me the warning message below. I am forced to reassign the prefab from the project folder to get it working. Is there any easier workaround? Otherwise it will be quite cumbersome to do this for all hand placed vegetation.
![image](https://github.com/visualdesigncafe/nature-renderer/assets/37171772/c63d7cb1-d951-45cd-8f0b-5764404f9c01)

**Comment by Ghostdreamer11** [comment]
AFAIK, you shouldn't assign itself.  However, I can't see a big reason why you 'cant' . The warning is just that --- a warning. If it renders fine, then it's fine. I am not sure what other performance implications you may be incurring by doing it this way though.

The nature instance stores a reference to the prefab it is instancing. In the case of converting prefabs present in the scene to nature instances, you can multi-select the prefabs in your scene and then in the Hierarchy right click > Convert to Nature Instance. 

At that point you can leave it as is, or prefab the nature instance that was created for continued development.

**Comment by maxplooi** [comment]
The Nature Instance component should be on an **empty game object in the scene**, then you can assign the Prefab that you want to render in the component.

All the rendering is handled by that single Nature Instance component, so having other components on the same object (like a Mesh Renderer) will result in double rendering. Once by the Nature Instance component and once by Unity via the Mesh Renderer component. You'll get the same issue when adding the component to the prefab asset as you did. Everything will be rendered twice.

If you have existing prefab instances in the scene then it is best to do what @Ghostdreamer11 said and convert them by right clicking and selecting **Convert to Nature Instance**.
